03/app.js

```diff
@@ -3,19 +3,57 @@ import ReactDOM from 'react-dom';
 
 import Category from './Category';
 import Cart from './Cart';
+import Product from './product.js'
 
 import data from './data.json';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
-    
+    list = (array, btnDel, func) => {
+        return array.map((d) => {
+
+        const id = d.id;
+
+        return <Product d={d} btn={btnDel} func={func} id={id}/>
+    })
+    }
+    addToCart = (e) => {
+        e.target.disabled = true;
+        data.forEach((d) => {
+            
+            if(e.target.parentElement.id == d.id ) {
+                return this.setState({cart: [...this.state.cart, d]})
+            }
+        })
+    }
```

Comment:
To podejście jest prawidłowe w JS, w React nie powinieneś się opierać na danych z DOM.
Napisz sobie funkcję `addToCart()` oraz `removeFromCart()`, do których przekazujesz `id` produktu.
W momencie renderowania produktu po prostu przekazujesz odpowiednią metodą np. `<Product handleClick={addToCart}>`. lub `<Product handleClick={removeFromCart}>` .
Natomiast wew. `<Product>` robisz `<button onClick={e => this.props.handleClick(this.props.id)}>{this.props.text}</button>`.
Dzięki takiemu rozwiązaniu wszystko dzieje się wew. komponentów i nie narażamy się na zmiany w DOM, które może wprowadzić użytkownik.